bin/npg_upload2climb

```diff
@@ -0,0 +1,274 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use FindBin qw($Bin);
+use lib ( -d "$Bin/../lib/perl5" ? "$Bin/../lib/perl5" : "$Bin/../lib" );
+use English qw(-no_match_vars);
+use Carp;
+use Readonly;
+use Getopt::Long;
+use Pod::Usage;
+use Log::Log4perl qw(:easy);
+use Try::Tiny;
+use Text::CSV;
+use File::Path qw(make_path);
+use File::Basename;
+use File::Spec::Functions;
+
+use npg_pipeline::function::pp_archiver;
+
+our $VERSION = '0';
+
+Readonly::Scalar my $MANIFEST_EXTENTION    => q[.tsv];
+Readonly::Scalar my $MANIFEST_SUFFIX       => q[.climbupload];
+Readonly::Scalar my $SHIFT_EIGHT           => 8;
+Readonly::Scalar my $UPLOAD_DIR            => q[upload];
+Readonly::Scalar my $UPLOADED_SAMPLES_GLOB => join q[/], $UPLOAD_DIR, q[*], q[];
+
+my $help;
+my $host;
+my $user;
+my $pkey_file;
+my $manifest_file;
+my $dry_run;
+
+GetOptions ('help'        => \$help,
+            'dry_run!'    => \$dry_run,
+            'host=s'      => \$host,
+            'user=s'      => \$user,
+            'pkey_file=s' => \$pkey_file,
+            'manifest=s'  => \$manifest_file,) or pod2usage(2);
+
+$help and pod2usage(1);
+
+Log::Log4perl->easy_init({layout => '%d %-5p %c - %m%n',
+                          level  => $INFO});
+my $logger = Log::Log4perl->get_logger();
+
+if (not -f $manifest_file) {
+  $logger->error("Manifest $manifest_file does not exist");
+  exit 1;
+}
+
+my $manifest4upload_file = $manifest_file;
+if ( not $manifest4upload_file =~ s/$MANIFEST_EXTENTION\Z/${MANIFEST_SUFFIX}${MANIFEST_EXTENTION}/xms) {
+  $manifest4upload_file .= $MANIFEST_SUFFIX . $MANIFEST_EXTENTION;
+}
+
+my $to_upload = [];
+my $e;
+try {
+  if (-e $manifest4upload_file) {
+    $logger->info("Found existing manifest for upload $manifest4upload_file");
```

Comment:
Minor variant: What about _always_ running the code in this block to read the upload manifest, and moving the upload manifest generation above (conditional on file not existing, and not filling in the `$to_upload` directely) - so there is only one code path for filling in `$to_upload`.

Reply:
Implemented; was leaning to this myself